plugins/inputs/consul/consul.go

```diff
@@ -0,0 +1,174 @@
+  ## Optional URI scheme for the Consul server (default: "http")
+  # scheme = "http"
+  ## Optional ACL token used in every request (default: "")
+  # token = ""
+  ## Optional username used for request HTTP Basic Authentication (default: "")
+  # username = ""
+  ## Optional password used for HTTP Basic Authentication (default: "")
+  # password = ""
+  ## Optional data centre to query the health checks from (default: "")
+  # datacentre = ""
+  ## Optional list of fields to be returned. Valid fields are: "check_id", "check_name", "node",
+  ##  "service_id", "service_name", "status", "notes", "output" (default: ["check_id", "check_name", "node",
+  ##  "service_id", "service_name", "status"])
+  # fields = ["check_id", "check_name", "service_id", "status"]
+  ## Optional list of field names to be used as tags (default: ["node", "service_name"])
+  # tags = ["node", "service_name"]
+`
+
+func (c *Consul) Description() string {
+	return "Gather health check statuses from services registered in Consul"
+}
+
+func (c *Consul) SampleConfig() string {
+	return sampleConfig
+}
+
+func (c *Consul) setDefaults() {
+	if len(c.Fields) == 0 {
+		c.Fields = []string{"check_id", "check_name", "service_id", "status"}
+	}
+
+	if len(c.Tags) == 0 {
+		c.Tags = []string{"node", "service_name"}
+	}
+}
+
+func (c *Consul) createAPIClient() (*api.Client, error) {
+	config := api.DefaultConfig()
+
+	if c.Address != "" {
+		config.Address = c.Address
+	}
+
+	if c.Scheme != "" {
+		config.Scheme = c.Scheme
+	}
+
+	if c.Datacentre != "" {
+		config.Datacenter = c.Datacentre
+	}
+
+	if c.Username != "" {
+		config.HttpAuth = &api.HttpBasicAuth{
+			Username: c.Username,
+			Password: c.Password,
+		}
+	}
+
+	tlsCfg, err := internal.GetTLSConfig(
+		c.SSLCert, c.SSLKey, c.SSLCA, c.InsecureSkipVerify)
+
+	if err != nil {
+		return nil, err
+	}
+
+	config.HttpClient.Transport = &http.Transport{
+		TLSClientConfig: tlsCfg,
+	}
+
+	return api.NewClient(config)
+}
+
+func (c *Consul) GatherHealthCheck(acc telegraf.Accumulator, checks []*api.HealthCheck) {
+	for _, check := range checks {
+		ref := reflect.Indirect(reflect.ValueOf(check))
+		record := make(map[string]interface{})
+		tags := make(map[string]string)
+
+		for _, field := range c.Fields {
+			value := ref.FieldByName(fieldMap[field])
+
+			if value.IsValid() {
+				record[field] = value.String()
+			}
+		}
+
+		if len(record) == 0 {
+			continue
+		}
+
+		for _, field := range c.Tags {
+			value := ref.FieldByName(fieldMap[field])
+
+			if value.IsValid() {
+				tags[field] = value.String()
+			}
+		}
+
+		acc.AddFields("consul", record, tags)
+	}
+}
+
+func (c *Consul) Gather(acc telegraf.Accumulator) error {
+	client, err := c.createAPIClient()
+
+	if err != nil {
+		return err
+	}
+
+	checks, _, err := client.Health().State("any", nil)
+
+	if err != nil {
+		return err
+	}
+
+	c.GatherHealthCheck(acc, checks)
+
+	return nil
+}
+
+func init() {
+	inputs.Add("consul", func() telegraf.Input {
```

Comment:
Would you mind renaming this to consul_health or something else specific to the endpoint? I'm working on a consul input for the agent endpoints and would be happy to combine it with this one once it's merged. That is, unless @sparrc perfers otherwise.